fcrepo-kernel-impl/src/test/java/org/fcrepo/kernel/impl/services/AbstractServiceTest.java

```diff
@@ -0,0 +1,237 @@
+/*
+ * Licensed to DuraSpace under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional information
+ * regarding copyright ownership.
+ *
+ * DuraSpace licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except in
+ * compliance with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.fcrepo.kernel.impl.services;
+
+
+import static org.fcrepo.kernel.api.RdfLexicon.BASIC_CONTAINER;
+import static org.fcrepo.kernel.api.RdfLexicon.CONTAINER;
+import static org.fcrepo.kernel.api.RdfLexicon.DEFAULT_INTERACTION_MODEL;
+import static org.fcrepo.kernel.api.RdfLexicon.FEDORA_RESOURCE;
+import static org.fcrepo.kernel.api.RdfLexicon.LDP_NAMESPACE;
+import static org.fcrepo.kernel.api.RdfLexicon.MEMENTO_TYPE;
+import static org.fcrepo.kernel.api.RdfLexicon.NON_RDF_SOURCE;
+import static org.fcrepo.kernel.api.RdfLexicon.REPOSITORY_NAMESPACE;
+import static org.fcrepo.kernel.api.RdfLexicon.RESOURCE;
+import static org.junit.Assert.assertEquals;
+
+import org.apache.commons.io.IOUtils;
+import org.apache.jena.rdf.model.Model;
+import org.apache.jena.rdf.model.ModelFactory;
+import org.apache.jena.vocabulary.DC_11;
+import org.apache.jena.vocabulary.XSD;
+import org.fcrepo.kernel.api.exception.RequestWithAclLinkHeaderException;
+import org.fcrepo.kernel.api.exception.ServerManagedTypeException;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.junit.MockitoJUnitRunner;
+
+import java.io.InputStream;
+import java.util.List;
+import java.util.UUID;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+@RunWith(MockitoJUnitRunner.Silent.class)
+public class AbstractServiceTest {
+
+    private final List<String> stringTypesNotValid = Stream.of(MEMENTO_TYPE, RESOURCE.toString(),
+            FEDORA_RESOURCE.toString()).collect(Collectors.toList());
+
+    private final List<String> stringTypesValid = Stream.of(MEMENTO_TYPE, CONTAINER.toString(),
+            BASIC_CONTAINER.toString()).collect(Collectors.toList());
+
+    private final String defaultInteractionModel = DEFAULT_INTERACTION_MODEL.toString();
+
+    private final AbstractService service = new AbstractService();
```

Comment:
`AbstractService` should be... `abstract` :) These tests are valuable, but they should be run against concrete implementations. You could move these test methods to test classes for specific services where they are applicable (for instance, interaction model stuff only really applies to creating resources). If there are any that apply to all services, then you could make this into a parent class for the other test classes, in which case you'll have to have an instantiation of the service available to these tests.